scripts/install-darwin-multi-user.sh

```diff
@@ -4,7 +4,11 @@ set -eu
 set -o pipefail
 
 # System specific settings
-export NIX_FIRST_BUILD_UID="${NIX_FIRST_BUILD_UID:-30001}"
+if [[ $(sw_vers -productVersion | cut -d '.' -f 1) -ge 15 ]]; then
```

Comment:
FYI, this is a very brittle check -- macOS sometimes decides to lie about the version for "compatibility". If this approach is accepted, you'll need to set `SYSTEM_VERSION_COMPAT=0` in this invocation (see e.g. https://eclecticlight.co/2020/08/13/macos-version-numbering-isnt-so-simple/) to ensure you're getting the actual version and not the compatibility version.